### PR TITLE
[PyHDK] Forwarding import of parquet files from cpp

### PR DESCRIPTION
**Pull request by Devjiu**
This commit supports importing parquet files via the Python API.

Resolves: #290

Signed-off-by: Dmitrii Makarenko <dmitrii.makarenko@intel.com>

### CODE REVIEW

File: python/tests/test_pyhdk_api.py [hunk 1]
```diff
@@ -130,6 +130,36 @@ def test_import_csv(self, header, create_table, full_schema, glob):
...
+    def test_import_parquet(self, create_table, glob):
+        hdk = pyhdk.init()
+        table_name = "table1"
+
+        if create_table:
+            hdk.create_table(table_name, (("a", "int64"), ("b", "fp64")))
+            real_schema = {"a": "INT64", "b": "FP64"}
+
+        if glob:
+            file_name = ["omniscidb/Tests/ArrowStorageDataFiles/int_float.parquet", "omniscidb/Tests/ArrowStorageDataFiles/int_float.parquet"]
```
Comment (ienkovich) [inline python/tests/test_pyhdk_api.py:144]: This doesn't actually a glob test. You should use some file pattern to test it.

File: python/tests/test_pyhdk_api.py [hunk 2]
```diff
@@ -130,6 +130,36 @@ def test_import_csv(self, header, create_table, full_schema, glob):
...
+            ref_data = {
+                "a": [*range(1, 6), *range(1, 6)],
+                "b": [1.1, 2.2, 3.3, 4.4, 5.5, 1.1, 2.2, 3.3, 4.4, 5.5],
+            }
+        else:
+            file_name = "omniscidb/Tests/ArrowStorageDataFiles/int_float.parquet"
+            ref_data = {"a": [*range(1, 6)], "b": [1.1, 2.2, 3.3, 4.4, 5.5]}
+
+        if not create_table:
```
Comment (ienkovich) [inline python/tests/test_pyhdk_api.py:153]: Either move it up to the previous `create_table` if-statement or define schema unconditionally

**Review by ienkovich — CHANGES_REQUESTED** [review]
The patch looks good! I just have some minor improvement suggestions.

File: python/pyhdk/hdk.py [hunk 3]
```diff
@@ -1853,7 +1852,47 @@ def import_parquet(self, file_name, table_name=None, fragment_size=None):
         QueryExpr
             Scan expression referencing created table.
         """
-        pass
+        files = []
```
[the remark this answers is not in the document]
Comment (Devjiu) [reply python/pyhdk/hdk.py]: done, check pls.

File: python/pyhdk/hdk.py [hunk 4]
```diff
@@ -1853,7 +1852,47 @@ def import_parquet(self, file_name, table_name=None, fragment_size=None):
...
+            raise RuntimeError(
+                f"Cannot resolve specified file name to any existing file."
+            )
+
+        table_opts = TableOptions()
+        if fragment_size is not None:
+            table_opts.fragment_size = fragment_size
+
+        real_table_name, append = self._process_import_table_name(table_name)
+        # Create new table by importing the forst file or simply check its schema
```
[the remark this answers is not in the document]
Comment (Devjiu) [reply python/pyhdk/hdk.py]: done.

File: python/tests/test_pyhdk_api.py (hunk 1 above)
Comment (Devjiu) [reply python/tests/test_pyhdk_api.py]: changed test. We should clarify supported types and conversion mapping in further work.

File: python/pyhdk/hdk.py [hunk 5]
```diff
@@ -1853,7 +1856,30 @@ def import_parquet(self, file_name, table_name=None, fragment_size=None):
...
+            table_opts.fragment_size = fragment_size
+
+        real_table_name, append = self._process_import_table_name(table_name)
+        # Create new table by importing the first file or simply check its schema
+        # matches the specified one.
+        if append:
+            # check schema
+            pass
+        else:
+            self._storage.importParquetFile( files[0], real_table_name, table_opts)
```
Comment (ienkovich) [inline python/pyhdk/hdk.py:1872]: Looks like the code is not formatted. We do not force Python code formatting in CI, but please apply `black` formatter here.

**Review by ienkovich — COMMENTED** [review]
The code looks good. I just propose minor improvements not related to the parquet functionality.

File: python/pyhdk/hdk.py [hunk 6]
```diff
@@ -1727,6 +1727,27 @@ def drop_table(self, table):
             raise TypeError(
                 f"Only str and QueryNode scans are allowed for 'table' arg. Provided: {table}"
             )
+        
+    def __verify_files_param(self, file_name):
+        files = []
+        if isinstance(file_name, str):
+            files.extend(sorted(glob.glob(file_name)))
```
[the remark this answers is not in the document]
Comment (Devjiu) [reply python/pyhdk/hdk.py]: done.

**Review by ienkovich — APPROVED** [review]

**Pull request merged**